docs/source/tutorial/neighborloader.rst

```diff
@@ -0,0 +1,169 @@
+NeighborLoader
+=====================================
+``NeighborLoader`` is a data loader that performs neighbor sampling as introduced in the “Inductive Representation Learning on Large Graphs” paper.
+This loader allows for mini-batch training of GNNs on large-scale graphs where full-batch training leads to out of memory problem.
```

Comment:
Can we provide a bit more detail about
* why we need sampling in general
* how it works on paper